src/kotlin_language.cpp

```diff
@@ -377,9 +379,12 @@ void KotlinLanguage::frame() {
 }
 
 bool KotlinLanguage::handles_global_class_type(const String& p_type) const {
-    return ScriptLanguage::handles_global_class_type(p_type);
+    return p_type == "KotlinScript";
 }
 
 String KotlinLanguage::get_global_class_name(const String& p_path, String* r_base_type, String* r_icon_path) const {
-    return ScriptLanguage::get_global_class_name(p_path, r_base_type, r_icon_path);
+    if (p_path.begins_with(GODOT_ENTRY_PATH)) {
```

Comment:
Implementing this methods make engine slow, we need to investigate this.  
Note to myself: If only used from editor, not include it in export.

Reply:
Do you want to have a look in this PR or do you want to tackle that in a separate PR?

Reply:
I think we should have a look here, as this introduce an unexpected behaviour

Reply:
not anymore slow engine closing, don't know what happened.